components/graphs/Graph.tsx

```diff
@@ -8,19 +8,26 @@ import {
   YAxis,
 } from 'recharts';
 import { LegendLineGraphIcon } from '@openclimatefix/nowcasting-ui.icons.icons';
-import { formatter } from 'lib/utils';
+import { formatter, forecastDataOverDateRange } from 'lib/graphs';
 import { useSiteData } from 'lib/hooks';
+import useTime from '~/lib/hooks/useTime';
 import { FC } from 'react';
 
 const Graph: FC<{ siteUUID: string }> = ({ siteUUID }) => {
-  const { forecastData } = useSiteData(siteUUID);
+  const { forecastData, latitude, longitude } = useSiteData(siteUUID);
+  const { currentTime } = useTime(latitude, longitude);
 
-  const maxGeneration = forecastData
-    ? Math.max(
-        ...forecastData.forecast_values.map(
-          (value) => value.expected_generation_kw
-        )
+  const endDate = new Date();
+  endDate.setHours(endDate.getHours() + 48);
+  const graphData = forecastData
+    ? forecastDataOverDateRange(
+        forecastData.forecast_values,
+        new Date(currentTime),
+        endDate
       )
+    : [];
```

Comment:
if setting this to [] by default breaks things (as in our bug with `ThresholdGraph`), perhaps we could consider just setting this to be `undefined` and checking for that instead of having to do `graphData.length > 0`?

Reply:
Just tried this out and the threshold label and current time marker are still not rendering

Reply:
alright keep it like this then